WcaOnRails/app/models/registration.rb

```diff
@@ -122,8 +122,11 @@ def waiting_list_info
 
   validate :user_can_register_for_competition
   private def user_can_register_for_competition
-    if user&.cannot_register_for_competition_reasons.present?
```

Comment:
I think you don't need to change the conditional that much, `present?` checks if the array is empty, so you should be able to just use:
```ruby
errors.add(:user_id, user.cannot_register_for_comptition_reasons.to_sentence)
```
instead of the previous error message.